kotlin-options-rxjava2-extensions/src/main/kotlin/com/petertackage/kotlinoptions/RxJava2Extensions.kt

```diff
@@ -6,57 +6,81 @@ import io.reactivex.Observable
 import io.reactivex.Single
 
 /*
- * filterIfSome
+ * filterNotNone
  */
 
-fun <T : Any> Observable<Option<T>>.filterIfSome(): Observable<T> {
-    return filter { it is Some }.map { it.getUnsafe() }
-}
 
-fun <T : Any> Observable<Option<T>>.filterIfSome(predicate: (T) -> Boolean): Observable<T> {
-    return filterIfSome().filter(predicate)
-}
 
-fun <T : Any> Flowable<Option<T>>.filterIfSome(): Flowable<T> {
-    return filter { it is Some }.map { it.getUnsafe() }
-}
+fun <T : Any> Observable<Option<T>>.filterNotNone(): Observable<T> =
```

Comment:
Could you add a comment block separating these by their Observable type (Observable, Single, Maybe etc) - it's quite hard reading at the moment.